pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +141,99 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
 	return nil
 }
 
+func (s secretResource) reconcileKeystoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsEncryptionEnabled() {
+		return nil, nil
+	}
+
+	keystoreSecret := &corev1.Secret{}
+	if result, err := kube.LookupResource(i.GetKeystoreSecretName(), i.Namespace, keystoreSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), s.client, keystoreSecret, func() error {
+		if ownerRef := metav1.GetControllerOf(keystoreSecret); ownerRef != nil && ownerRef.UID != s.infinispan.UID {
+			return fmt.Errorf("Unable to SetControllerReference for secret %s as it's already owned  by %s:%s:%s",
```

Comment:
Can we make this error more customer facing. `SetControllerReference` a bit internal knowledge.
Also would be nice to send events for this kind of errors/warnings.

Reply:
> Can we make this error more customer facing. `SetControllerReference` a bit internal knowledge.

How about:

"Unable to update Secret %s's ownerReference with controller=true asit's already owned by %s:%s:%s" ?

I can update this to send an event once #1008 has been merged.